### DISCUSSION TITLE
If two shapes overlap when touching the ground, only one should drop

### DISCUSSION THREAD

**Discussion by JarodStreck** [web-classroom/web-classroom.github.io#6] [Q&A : Pratique / Labos] [answered]
Si j'ai bien compris, pour ce test, si 2 shapes sont l'une sur l'autre pendant qu'elles tombent et touchent le sol en même temps **et au même endroit** seulement l'une des 2 doit être drop et grounded.

Quel est le comportement attendu pour la deuxième Shape ?

**Comment by OlivierLmr** [comment, marked as answer]
Ce test vérifie la phrase suivante de la section [Game Step](https://web-classroom.github.io/labos/labo-2-tetris-1.html#game-step) :

> Si plusieurs pièces doivent être posée au même tour mais se superposent, l’une d’elles est choisie arbitrairement pour être posée, et l’autre est remplacée.

Ça signifie donc que si, à un même game step, deux shapes ne peuvent pas descendre d'un cran et doivent donc être posées, mais qu'elles se superposent (donc elles ne sont pas nécessairement exactement à la même position, mais au moins un de leurs blocs overlap), alors une seule d'entre elles doit effectivement être posée (laquelle, ça n'a pas d'importance), et l'autre doit être supprimée et remplacée par une nouvelle pièce en haut du board.

En gros, si deux shapes doivent être posées en même temps mais que c'est impossible parce qu'il y a au moins un bloc sur lequel elles se superposent, alors une seule doit être effectivement posée, et l'autre doit etre remplacée.
